scala/Scala.g4

```diff
@@ -157,17 +161,25 @@ infixExpr
    ;
 
 prefixExpr
-   : ('-' | '+' | '~' | '!')? ('new' (classTemplate | templateBody) | blockExpr | simpleExpr1 '_'?)
+   : ('-' | '+' | '~' | '!')? (simpleExpr | simpleExpr1)
+   ;
+
+simpleExpr
+   : 'new' (classTemplate | templateBody)
+   | blockExpr
    ;
 
+// Dublicate lines to prevent left-recursive code.
+// can't use (simpleExpr|simpleExpr1) '.' Id
 simpleExpr1
    : literal
-   | stableId
-   | (Id '.')? 'this'
+   | path
    | '_'
    | '(' exprs? ')'
-   | ('new' (classTemplate | templateBody) | blockExpr) '.' Id
-   | ('new' (classTemplate | templateBody) | blockExpr) typeArgs
+   | simpleExpr  '.' Id
+   | simpleExpr1 '.' Id
+   | simpleExpr  typeArgs
+   | simpleExpr1 typeArgs
```

Comment:
```g4
    | (simpleExpr | simpleExpr1) '.' Id
    | (simpleExpr | simpleExpr1) typeArgs
```

Reply:
Can't do this. You can try it, but the result is
`error(119): Scala.g4::: The following sets of rules are mutually left-recursive [simpleExpr1]`

That's why I split it to a different line. I think it some problem in generator.

Cause 
`| (simpleExpr | simpleExpr1) '.' Id` 
equal to 
`| simpleExpr '.' Id | simpleExpr1 '.' Id`

Reply:
Yes, it takes place if left-recursion is involved.

Reply:
But why, logical it's the same rule. Looks like just a trick to hide from some check in antlr core. Maybe investigate?)

Reply:
Already done. There are several issues in ANTLR repository that are similar to our case, for instance, this: https://github.com/antlr/antlr4/issues/564